ui/helpers/utils/util.js

```diff
@@ -715,6 +716,27 @@ export const isAbleToExportAccount = (keyringType = '') => {
   return !keyringType.includes('Hardware') && !keyringType.includes('Snap');
 };
 
+export const isAbleToRevealSrp = (accountToExport, keyrings) => {
+  if (
+    accountToExport.metadata.keyring.type !== KeyringTypes.hd &&
+    accountToExport.metadata.keyring.type !== KeyringTypes.snap
+  ) {
+    return false;
+  }
+
+  // All hd keyrings can reveal their srp.
+  if (accountToExport.metadata.keyring.type === KeyringTypes.hd) {
+    return true;
+  }
+
+  // For snap accounts we must check if the entropy source was from a HD keyring.
+  const keyringId = accountToExport.metadata.keyring.id;
```

Comment:
Nit: Cosmetic:

We could have de-structured the `keyring.type` and `keyring.id` there.